docs/developing-operators.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Developing operators
+type: docs
+menu: docs
+---
+
+This guide will provide introduction to creating KUDO operators, you will learn about the structure of the package and the template language to use.
+
+## Getting started
+In this section we’ll start by developing your first operator and we’ll follow up with in-depth explanation of the underlying concepts.
+
+The overall structure of a package looks following:
+```shell
+.
+├── operator.yaml
+├── params.yaml
+└── templates
+    ├── deployment.yaml
+    └── ...
+```
+
+The `operator.yaml` is the main yaml file defining both operator metadata as the whole lifecycle of the operator. `params.yaml` defines parameters of the operator. During installation, these parameters can be overriden allowing customization. `templates` folder contain all templated kubernetes objects that will be applied to your cluster after installation based on the workflow defined in `operator.yaml`.
+
+### Your first KUDO operator
+First let’s create `operator.yaml` and place it in a `first-operator` folder.
+
+```shell
+name: "first-operator"
+version: "0.1.0"
+maintainers:
+- Your name <your@email.com>
+url: https://kudo.dev
+tasks:
+  nginx:
+    resources:
+      - deployment.yaml
+plans:
+  deploy:
+    strategy: serial
+    phases:
+      - name: main
+        strategy: parallel
+        steps:
+          - name: everything
+            tasks:
+              - nginx
+```
+
+As you can see we have operator with just one plan - deploy, with one phase and one step which is the minimal setup. Deploy plan is automatically triggered when you install instance of this operator into cluster.
```

Comment:
this is really hard for a noob to follow IMO.   We can't explain everything at once and I like the abbreviated tutorial.   There needs to be either 1) more explanation around plan or 2) a link to more detail around plans.

The in-depth view of plans, phases and steps should be covered in a separate doc IMO.   A brief mention here would be good.   

```
Kudo operators may be configured with multiple plans.  The default expected plan is the `deploy` plan which will be triggered when you install an instance of this operator.  Each plan has a series of phases which each contain a series of steps.  This example deploy plan has 1 phase named `main` and it has 1 step named `everything`.  For more details on plans see...
```

Reply:
I wrote it in description. This is not final draft, I want to move iteratively not write 10 pages of docs at once. This will come in later PRs. This PR does not close the whole documentation issue.